main/main.ino

```diff
@@ -970,6 +970,27 @@ bool wifi_reconnect_bypass() {
   while (WiFi.waitForConnectResult() != WL_CONNECTED && wifi_autoreconnect_cnt < maxConnectionRetryWifi) {
 #  endif
     Log.notice(F("Attempting Wifi connection with saved AP: %d" CR), wifi_autoreconnect_cnt);
+
+    //Reduce WiFi interference when using ESP32 using custom WiFi mode and tx power
+#  ifdef ESP32
+    // https://github.com/jnogues/ESP32-Long-Range-WiFi/issues/4
+    // Reset wifi protocol
+    if (esp_wifi_set_protocol(WIFI_IF_STA, WIFI_PROTOCOL_11B | WIFI_PROTOCOL_11G | WIFI_PROTOCOL_11N | WIFI_PROTOCOL_LR) != ESP_OK) {
```

Comment:
The goal of this is to improve wifi reconnection ?

Reply:
As I understand it when we start changing the Wifi band it gets stored in the ESP32 firmware and needs to be reset to default using explicit code.  Hence the research and reference to https://github.com/jnogues/ESP32-Long-Range-WiFi/issues/4 where other Platform.IO users had the same experience.

I could not understand why my ESP32 would not want to connect to Band N after lowering it to Band G even after removing my code. And this fixed it.

Reply:
I just tested it, and it prevents the wifi configuration portal from starting

Reply:
Busy testing a solution.

Reply:
If we could avoid going through this function for regular users we will save a bunch of testing and potential headache

Reply:
I got rid of that; from my testing, it should be working, incl. WiFi configuration.  I did an {"cmd":"erase"} and checked that I can connect and reconfigure.

Reply:
See whether the latest changes are more suitable; I have done quite a bit of testing and it seems fine.

Reply:
I'm not a fan of impacting the wifi connection for everybody. We had users struggling with their routers the last time we did that.
Can these changes be only for HC-SR501?

Reply:
The latest pre-processor statements prevent this code from impacting existing users.  Only when they **explicitly**  **DEFINE** or **uncomment** either **WifiGMode** or **WifiPower** and have an **ESP32** will the code be compiled which changes the WiFi protocol and transmit power.

Reply:
Great!